src/resolve-config.js

```diff
@@ -26,23 +28,43 @@ function getLoadFunction(opts) {
   return getExplorerMemoized(opts).load;
 }
 
-function resolveConfig(filePath, opts) {
+function _resolveConfig(filePath, opts, sync) {
   opts = Object.assign({ useCache: true }, opts);
-  const load = getLoadFunction({ cache: !!opts.useCache, sync: false });
-  return load(filePath, opts.config).then(result => {
-    return !result ? null : mergeOverrides(result, filePath);
-  });
+  const loadOpts = { cache: !!opts.useCache, sync: !!sync };
+  const load = getLoadFunction(loadOpts);
+  const loadEditorConfig = resolveEditorConfig.getLoadFunction(loadOpts);
+  const arr = [load, loadEditorConfig].map(l => l(filePath, opts.config));
+
+  const unwrapAndMerge = arr => {
+    const result = arr[0];
+    const editorConfigured = arr[1];
+    const merged = Object.assign(
+      {},
+      editorConfigured,
+      mergeOverrides(Object.assign({}, result), filePath)
+    );
+
+    if (Object.keys(merged).length === 0) {
```

Comment:
This doesn't look the same as the current behaviour. We want an empty config file to be different than no config file. How this works with `.editorconfig`... I don't know. Editors want to support formatting only if a config file is found, so supporting EditorConfig might break that functionality.

Reply:
> This doesn't look the same as the current behaviour.

According to @robwise in https://github.com/prettier/prettier/pull/2760#discussion_r145477197, the current behavior treats missing/empty config files equivalently. I wanted to be sure, so here's a script you can run against a clean build of cbed0f487f2f69b8f8a70ba73ec7115762556cfb to try it out:

```bash
tmpdir=`mktemp -d`
tmpfile=$tmpdir/file.js
tmprc=$tmpdir/.prettierrc

echo -n > $tmpfile

echo "With no .prettierrc"
node -p "require('./').resolveConfig.sync('$tmpfile')"
echo

echo "With empty .prettierrc"
echo -n > $tmprc
node -p "require('./').resolveConfig.sync('$tmpfile')"
echo

rm -rf $tmpdir
```

On my machine, this prints:

```
With no .prettierrc
null

With empty .prettierrc
null
```

Because of this, I'd like to fix this missing/empty distinction in a separate PR, since it's pre-existing and (somewhat?) unrelated (see https://github.com/prettier/prettier/pull/2760#discussion_r146105548)

> How this works with .editorconfig... I don't know. Editors want to support formatting only if a config file is found, so supporting EditorConfig might break that functionality.

Hmm, I guess the use-case here is:

* The user has a prettier integration in their editor
* The integration is configured to format-on-save if a prettier configuration exists
* The user is working on a project that has a .editorconfig
* But the project's style guide is incompatible with prettier's formatting

Yeah, I could see how that might get tricky. I'm going to have to give it some more thought.

Reply:
> Yeah, I could see how that might get tricky. I'm going to have to give it some more thought.

Reply:
Yeah, I could see how that might get tricky. I'm going to have to give it some more thought.

Reply:
> Yeah, I could see how that might get tricky. I'm going to have to give it some more thought.

Ok, so one way to fix this is to make it so that `resolveConfigFile()` (and its CLI counterpart `--find-config-path`) return paths to empty `.prettierrc` files (it doesn't currently). Once that's the case, the editor integration can use it to determine whether an empty `.prettierrc` exists, even if `resolveConfig` returns an object derived from `.editorconfig`.

Reply:
Using `resolveConfigFile(path) != null` could be a solution.
I've not seen this function exposed for now, seems to be cli only.

Reply:
> Ok, so one way to fix this is to make it so that resolveConfigFile() (and its CLI counterpart --find-config-path) return paths to empty .prettierrc files (it doesn't currently).

Hmm, it looks like cosmiconfig intentionally ignores empty files: https://github.com/davidtheclark/cosmiconfig/commit/1adf5c2f10112322e12d53f7f779d559aef5c627

However, a `.prettierrc` file containing just a space or a newline is still found, so that could be the suggested workaround instead of this from https://github.com/prettier/prettier/pull/2760#discussion_r145477197: